src/components/MessageList/__tests__/MessageList.test.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { cleanup, render, waitFor } from '@testing-library/react-native';
+import {
+  dispatchMessageNewEvent,
+  generateChannel,
+  generateMember,
+  generateMessage,
+  generateStaticMessage,
+  generateStaticUser,
+  generateUser,
+  getOrCreateChannelApi,
+  getTestClientWithUser,
+  useMockedApis,
+} from 'mock-builders';
+
+import { Channel } from '../../Channel';
+import { Chat } from '../../Chat';
+import MessageList from '../MessageList';
+
+describe('MessageList', () => {
```

Comment:
Can we add some more tests?

I can think of following few so far!! This will make sure that custom components passed through props work properly:

- should render MessageSystem for message type `system`
- should render DateSeparator when messages differ in dates
- should render TypingIndicator when typing object is non-empty
- should render ErrorNotification when `online` prop is false

Also some related to scrolling (although I am not sure how easy is it to write test for scroll event. We can skip this if its too complicated)
- should render MessageNotification when new message is received and user has scroll up in list
- should remove MessageNotification when user scrolls down on list